homeassistant/components/devolo_home_control/switch.py

```diff
@@ -0,0 +1,155 @@
+"""Platform for light integration."""
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all devices and setup the switch devices via config entry."""
+    devices = hass.data[DOMAIN]["homecontrol"].binary_switch_devices
+
+    entities = []
+    for device in devices:
+        for binary_switch in device.binary_switch_property:
+            entities.append(
+                DevoloSwitch(
+                    homecontrol=hass.data[DOMAIN]["homecontrol"],
+                    device_instance=device,
+                    element_uid=binary_switch,
+                )
+            )
+    async_add_entities(entities)
+
+
+class DevoloSwitch(SwitchDevice):
+    """Representation of an Awesome Light."""
+
+    def __init__(self, homecontrol, device_instance, element_uid):
+        """Initialize an devolo Switch."""
+        self._device_instance = device_instance
+
+        # Create the unique ID
+        self._unique_id = element_uid
+
+        self._homecontrol = homecontrol
+        self._name = self._device_instance.itemName
+        self._available = self._device_instance.is_online()
```

Comment:
There's no I/O here, right?

Reply:
Correct, it is just read from the API internal data structure, which is kept in sync via a websocket connection